### ISSUE TITLE
Can't install modules in Ubuntu 15.10

### ISSUE THREAD

**Issue by atollkuci** [varnish/varnish-modules#42]
Hello,
I'm trying to install the varnish modules following the steps listed in documentation. All the dependencies are installed. I was also able to run the ./configure script, but it fails when I try to make. The error says:

_atollkuci@CACHE-SRV:~/varnish-4.0.3/varnish-modules$ make
make  all-recursive
make[1]: Entering directory '/home/atollkuci/varnish-4.0.3/varnish-modules'
Making all in src
make[2]: Entering directory '/home/atollkuci/varnish-4.0.3/varnish-modules/src'
/bin/bash: PYTHON@: command not found
Makefile:803: recipe for target 'vcc_cookie_if.h' failed
make[2]: *** [vcc_cookie_if.h] Error 127
make[2]: Leaving directory '/home/atollkuci/varnish-4.0.3/varnish-modules/src'
Makefile:485: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/atollkuci/varnish-4.0.3/varnish-modules'
Makefile:396: recipe for target 'all' failed
make: *** [all] Error 2_

Any help it's appreciated.

**Comment by Dridi** [comment]
This collection of modules is only supported for the latest version of Varnish, it won't work with varnish 4.0. Varnish 4.0 will reach its End Of Life in less than two weeks, I recommend to upgrade to 4.1 soon.

Closing this issue.

**Comment by atollkuci** [comment]
I did upgrade to latest version of varnish from github: varnishd (varnish-trunk revision c99382d)

Now I'm getting another error:

In file included from /usr/local/include/varnish/common/common.h:39:0,
                 from /usr/local/include/varnish/cache/cache.h:39,
                 from vmod_softpurge.c:30:
vmod_softpurge.c: In function 'vmod_softpurge':
/usr/local/include/varnish/vqueue.h:540:39: error: 'struct objcore' has no member named 'list'
 #define VTAILQ_NEXT(elm, field) ((elm)->field.vtqe_next)
                                       ^
/usr/local/include/varnish/vqueue.h:460:14: note: in expansion of macro 'VTAILQ_NEXT'
      (var) = VTAILQ_NEXT((var), field))
              ^
vmod_softpurge.c:63:2: note: in expansion of macro 'VTAILQ_FOREACH'
  VTAILQ_FOREACH(oc, &oh->objcs, list) {
  ^
vmod_softpurge.c:68:23: error: 'OC_EF_DYING' undeclared (first use in this function)
   if (oc->exp_flags & OC_EF_DYING)
                       ^
vmod_softpurge.c:68:23: note: each undeclared identifier is reported only once for each function it appears in
vmod_softpurge.c:85:27: error: 'struct objcore' has no member named 'exp'
   EXP_Rearm(oc, now, 0, oc->exp.grace, oc->exp.keep);
                           ^
vmod_softpurge.c:85:42: error: 'struct objcore' has no member named 'exp'
   EXP_Rearm(oc, now, 0, oc->exp.grace, oc->exp.keep);

Is the latest version supported or should I stick to 4.1?

Thank you

**Comment by Dridi** [comment]
I recommend to stick with 4.1, but if you wish to run the Varnish master branch in production you can try this fork of varnish-modules:

https://github.com/nigoroll/varnish-modules/

Please open issues there if you run into problems against varnish-trunk, this project currently targets Varnish 4.1.

**Comment by atollkuci** [comment]
I should have read more carefully the documentation. With varnish 4.1 everything did go fine.

If I can recommend something, please make it clear in the doc that it only works with varnish 4.1.

Thank you

**Comment by Dridi** [comment]
I just looked at the README and among the non-goals it says:

> Support older releases of Varnish Cache.

Could be improved.
